tests/test_main.py

```diff
@@ -12,14 +12,20 @@ def test_main_with_t1oos_not_handled(tmp_path):
         main.main({"input": input_})
 
 
-def test_main_with_dummy_data_file(tmp_path):
+@pytest.mark.parametrize("dummy_data_file", ["dummy_data.csv", None])
+def test_main_with_and_without_dummy_data_file(tmp_path, dummy_data_file):
```

Comment:
I think we could spare the characters and have one test for when a dummy data file is supplied (`test_main_with_dummy_data_file`) and one when it isn't. Whilst we shouldn't be dogmatic about logic in tests, the logic in _this_ test obscures that in one branch, the result CSV contains two rows and in the other branch it contains one row.

If you're unconvinced, then think about what a failure would look like:

```
FAILED tests/test_main.py::test_main_with_and_without_dummy_data_file[None] - assert 'patient_id\n""\n' == 'patient_id\n'
```